TheaterSchedule/Controllers/AdminsPostController.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using TheaterSchedule.BLL.Interfaces;
+using TheaterSchedule.BLL.DTOs;
+
+namespace TheaterSchedule.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AdminsPostController : ControllerBase
+    {
+        private IAdminsPostService adminsPostService;
+
+        public AdminsPostController(IAdminsPostService adminsPostService)
+        {
+            this.adminsPostService = adminsPostService;
+        }
+
+        [HttpGet]
+        [Produces("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult<IEnumerable<AdminsPostDTO>> GetAllPublicPosts()
+        {
+            return Ok(adminsPostService.GetAllPublicPosts());
+        }
+
+        [HttpGet("{id}")]
+        [Produces("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult<IEnumerable<AdminsPostDTO>> GetAllPersonalPosts(int id)
+        {
+            return Ok(adminsPostService.GetAllPersonalPosts(id));
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public ActionResult<AdminsPostDTO> AddNewAdminsPost(AdminsPostDTO post)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+            try
+            {
+                adminsPostService.AddNewPost(post);
+                return StatusCode(StatusCodes.Status201Created, post);
+            }
+            catch (System.Exception e)
```

Comment:
don't catch. we have exception handling middleware

Reply:
![image](https://user-images.githubusercontent.com/43295127/61203851-38ff5780-a6f4-11e9-8cb5-cdf1b762fba6.png)
Without try/catch we have unhandled exception. It's may not occur in real life since I was testing a case far from usual. I'll delete try/catch